### PR TITLE
Create k8s cron jobs

### PR DESCRIPTION
**Pull request by psoleckimoj**

### CODE REVIEW

File: helm_deploy/help-with-prison-visits-asynchronous-worker/templates/daily-tasks-cron.yaml [hunk 1]
```diff
@@ -0,0 +1,22 @@
+apiVersion: batch/v1beta1
+kind: CronJob
+metadata:
+  name: {{ include "app.fullname" . }}-cron-daily-tasks
+  labels:
+    {{- include "app.labels" . | nindent 4 }}
+spec:
+  schedule: "30 4 * * *"
```
Comment (mattops) [inline helm_deploy/help-with-prison-visits-asynchronous-worker/templates/daily-tasks-cron.yaml:8]: Does this run at weekends also?

File: helm_deploy/help-with-prison-visits-asynchronous-worker/templates/daily-tasks-cron.yaml [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+  failedJobsHistoryLimit: 5
+  startingDeadlineSeconds: 600
+  successfulJobsHistoryLimit: 5
+  jobTemplate:
+    spec:
+      template:
+        spec:
+          containers:
+            - name: daily-tasks
+              image: quay.io/hmpps/help-with-prison-visits-asynchronous-worker
```
Comment (mattops) [inline helm_deploy/help-with-prison-visits-asynchronous-worker/templates/daily-tasks-cron.yaml:19]: might want to a `command:` option here as well just to be clear what is going to run.

File: helm_deploy/help-with-prison-visits-asynchronous-worker/templates/worker-tasks-cron.yaml [hunk 3]
```diff
@@ -0,0 +1,22 @@
+apiVersion: batch/v1beta1
+kind: CronJob
+metadata:
+  name: {{ include "app.fullname" . }}-cron-worker-tasks
+  labels:
+    {{- include "app.labels" . | nindent 4 }}
+spec:
+  schedule: "* * * * *"
```
Comment (mattops) [inline helm_deploy/help-with-prison-visits-asynchronous-worker/templates/worker-tasks-cron.yaml:8]: this is every minute?  How long does the job take to run?

File: helm_deploy/help-with-prison-visits-asynchronous-worker/templates/daily-tasks-cron.yaml (hunk 1 above)
Comment (psoleckimoj) [reply helm_deploy/help-with-prison-visits-asynchronous-worker/templates/daily-tasks-cron.yaml:8]: Apparently so

File: helm_deploy/help-with-prison-visits-asynchronous-worker/templates/worker-tasks-cron.yaml (hunk 3 above)
Comment (psoleckimoj) [reply helm_deploy/help-with-prison-visits-asynchronous-worker/templates/worker-tasks-cron.yaml:8]: It was actually set for every 5 seconds under the JS cron stuff as it allowed that.

File: helm_deploy/help-with-prison-visits-asynchronous-worker/templates/daily-tasks-cron.yaml (hunk 1 above)
Comment (rwhittlemoj) [reply helm_deploy/help-with-prison-visits-asynchronous-worker/templates/daily-tasks-cron.yaml:9]: is there a conflict between the job history limits and the ttlSecondsAfterFinished?  Won't this delete each job 5 mins after completion, so eventually you wont have much of a job history?  Especially considering this runs at 04:30 in the morning, by the time someone comes in to look at the logs, the job will have been cleaned up?

Comment (psoleckimoj) [reply helm_deploy/help-with-prison-visits-asynchronous-worker/templates/daily-tasks-cron.yaml:9]: Interesting, never thought of that :D

**Review by mattops — APPROVED** [review]

**Pull request merged**